src/metrics/peerstore.go

```diff
@@ -67,18 +67,41 @@ func (c *PeerStore) StoreOrUpdatePeer(peer Peer) {
 	if err != nil {
 		c.PeerStore.Store(peer.PeerId, peer)
 	} else {
-		// update only the following parameters
-		oldPeer.NodeId = peer.NodeId
-		oldPeer.UserAgent = peer.UserAgent
-		oldPeer.ClientName = peer.ClientName
-		oldPeer.ClientOS = peer.ClientOS
-		oldPeer.ClientVersion = peer.ClientVersion
-		oldPeer.Pubkey = peer.Pubkey
-		oldPeer.Addrs = peer.Addrs
-		oldPeer.Ip = peer.Ip
-		oldPeer.Country = peer.Country
-		oldPeer.City = peer.City
-		oldPeer.Latency = peer.Latency
+		// update only the following parameters only if the incoming info is full
+		if peer.NodeId != "" {
```

Comment:
perhaps we can write a utility function to avoid that many boilerplate ifs?

something like this, note that its some quick pseudocode.

```go
peer.UserAgent = getNotEmpty(peer.UserAgent, oldPeer.UserAgent)
```

```go
func getNotEmpty(a, b){
  if a == "" {
    return b
  }
return a
}
```

Not a priority, so as you wish

Reply:
I like how this looks